workers/scanner.go

```diff
@@ -399,6 +425,36 @@ func (s *Scanner) managedRefreshMinPinners() {
 	s.mu.Unlock()
 }
 
+// staticEligibleToPin returns true when this server is either pinning less than
+// AlwaysPinThreshold data or it's in the lower PinningRangeThresholdPercent
+// section of servers when ordered by contract data in descending order.
+// A server is always eligible if it's last in the list.
+func (s *Scanner) staticEligibleToPin(ctx context.Context) (bool, error) {
+	pinnedData, err := s.staticDB.ServerLoad(ctx, s.staticServerName)
+	if err != nil {
+		return false, err
+	}
+	// Below the hard limit.
+	if pinnedData < int64(AlwaysPinThreshold) {
+		return true, nil
+	}
+	pos, total, err := s.staticDB.ServerLoadPosition(ctx, s.staticServerName)
```

Comment:
(nit) if this also returned the pinned data, then you can fetch all the information you need from the database with a single read vs 2 interactions.

Reply:
You mean that I wouldn't need `ServerLoad`? Hm... On one hand one RT to the DB is better than two. On the other hand `ServerLoadPositions` returns all records in the table and also orders them by a field, which is a lot of work.

With the size of the table in question it's a toss-up between the two. I'll keep the status quo for the moment.